### DISCUSSION TITLE
Data Model

### DISCUSSION THREAD

**Discussion by eriicafes** [eriicafes/revision-study-api#2] [General]
We'll discuss all models and their relations to one another here

**Comment by princecodes247** [comment]
Ok cool

**Comment by princecodes247** [comment]
So the ideally, 

1. The user signs up 
2. Creates a new bank 
3. Add questions to the bank
4. Take tests using one or more banks

Right?

**Reply by eriicafes** [reply]
Almost

**Reply by eriicafes** [reply]
The user can take a test on one bank at a time, if he wants to take a test on more banks he'll have to create a **tag** which can contain many banks

**Reply by eriicafes** [reply]
Users can share their question banks or tags to other users

**Reply by princecodes247** [reply]
Oh, cool

So are the tags Persistent too?

**Reply by eriicafes** [reply]
Yes they'll be

**Reply by princecodes247** [reply]
What do you think?

User Table
Tag Table
Question table

**Reply by eriicafes** [reply]
Yes

Let me picture it

**Reply by princecodes247** [reply]
Or should the Question Table be Bank Table?

**Reply by eriicafes** [reply]
I dropped a tabular representation

**Comment by eriicafes** [comment]
Data Model

| Table   | Relations           |
| --------| --------------------|
| users   | (primary table)     |
| questions| belongs to one user, belongs to one bank           |
| banks    | belongs to one user, belongs to many tags  |
| tags     | belongs to one user, has many banks|

**Reply by princecodes247** [reply]
Describe the tables

**Reply by eriicafes** [reply]
**users**
This stores the user information we could also complement it with a profile table if we need to store additional information about the user

**questions**
This stores the questions
Each question will be assigned a user so we can query all the users questions and also cascade delete them when we drop the user
Each question will also be assigned to one bank so we can query questions by bank and cascade delete also

**banks**
This stores the question banks(we probably need to change the name)
The banks will have information of how the tests will be when taken, and how we sort the questions while fetching(for the test)

**tags**
This stores the tags
The relationship between this table and the banks table should be a many to many, because tags can have multiple questions and those questions can belong to multiple tags

**Reply by princecodes247** [reply]
Questions would belong to banks too right?

**Reply by eriicafes** [reply]
Yes, that relationship is stored on the question as in bank_id
